model/chunks/chunkassignment.go

```diff
@@ -4,21 +4,40 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// Assignment is assignment map of the chunks to the list of the verifier nodes
+// Assignment is an immutable list that, for each chunk (in order of chunk.Index),
+// records the set of verifier nodes that are assigned to verify that chunk.
 type Assignment struct {
-	// TODO: use a slice here instead of a map, which will be more performant
-	verifiersForChunk map[uint64]map[flow.Identifier]struct{}
+	verifiersForChunk []map[flow.Identifier]struct{}
 }
 
-func NewAssignment() *Assignment {
+// AssignmentBuilder is a helper to create a new Assignment
+type AssignmentBuilder struct {
+	verifiersForChunk []map[flow.Identifier]struct{}
+}
+
+func NewAssignmentBuilder() *AssignmentBuilder {
+	return &AssignmentBuilder{
+		verifiersForChunk: make([]map[flow.Identifier]struct{}, 0, 1),
+	}
+}
+
+// Build completes the assignment and makes it immutable.
+func (a *AssignmentBuilder) Build() *Assignment {
+	// TODO is this correct?
+	// the underlying data can still be modified by the Builder,
+	// but the Builder ideally shouldn't be kept around or reused, so it should be sufficient
 	return &Assignment{
-		verifiersForChunk: make(map[uint64]map[flow.Identifier]struct{}),
+		verifiersForChunk: a.verifiersForChunk,
 	}
 }
 
 // Verifiers returns the list of verifier nodes assigned to a chunk
 func (a *Assignment) Verifiers(chunk *flow.Chunk) flow.IdentifierList {
```

Comment:
In several existing functions (`Verifiers`, `HasVerifier`, and `Add`) we are passing in a `Chunk` but only using the `Index` field. Here, the documentation states `returns the list of verifier nodes assigned to a chunk`, but we aren't actually checking anything about the chunk besides its index. Really it's up to the caller to make sure the chunk and assignment are associated, otherwise we might return an incorrect verifier set, without an error indicating the failure. 

I think passing in just the index, rather than the entire chunk, would make this fact more obvious in the API. If you agree, do you mind making that change here as well?